File: frontend/src/core/services/iServices/IPictureService.ts
import type { PictureDto } from "@/core/models/dto/PictureDto";
import type { ResponseDto } from "@/core/models/dto/ResponseDto";

export interface IPictureService {
  get(): Promise<ResponseDto>;
  add(pictureDto: PictureDto): Promise<ResponseDto>;
}

File: Snp.Services.PictureApi/Snp.Services.PictureAlbumApi/Services/IServices/IPictureService.cs
using Snp.Services.PictureAlbumApi.Models.Dto;

namespace Snp.Services.PictureAlbumApi.Services.IServices
{
    public interface IPictureService
    {
        Task<IEnumerable<PictureListItemDto>> GetAsync();

        Task<string> AddAsync(PictureDto pictureDto);
    }
}
